### PR TITLE
TST/CLN: clean up indexes/multi/test_unique_and_duplicates

### PR DESCRIPTION
**Pull request by h-vetinari**
Splitting up #21645

* Added tests for `duplicated`, including a fixture for a `MultiIndex` with duplicates
* Broke up a huge test (`test_duplicates`) into smaller chunks
* removed a test (`test_unique_na`) that was for an `Index` (not `MultiIndex`), and exists verbatim in `tests/indexes/common.py`
* moved `test_duplicate_level_names_access_raises` to its appropriate module

### CODE REVIEW

**Comment by jbrockmendel** [comment]
Test failure appears unrelated.  Can you push again so it re-runs?

**Comment by h-vetinari** [comment]
@jbrockmendel All green :)

File: pandas/tests/indexes/multi/test_names.py [hunk 1]
```diff
@@ -115,3 +115,9 @@ def test_names(idx, index_names):
     ind_names = list(index.names)
     level_names = [level.name for level in index.levels]
     assert ind_names == level_names
+
+
+def test_duplicate_level_names_access_raises(idx):
+    idx.names = ['foo', 'foo']
```
Comment (jreback) [inline pandas/tests/indexes/multi/test_names.py]: is there an issue number asssociated?

File: pandas/tests/indexes/multi/test_unique_and_duplicates.py [hunk 2]
```diff
@@ -7,10 +7,24 @@
 import pandas as pd
 import pandas.util.testing as tm
 import pytest
-from pandas import MultiIndex
 from pandas.compat import range, u
 
 
+@pytest.fixture
```
Comment (jreback) [inline pandas/tests/indexes/multi/test_unique_and_duplicates.py]: should go in the conftest.py

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/indexes/multi/test_names.py (hunk 1 above)
Comment (h-vetinari) [reply pandas/tests/indexes/multi/test_names.py]: Don't know, just moving tests around

Comment (h-vetinari) [reply pandas/tests/indexes/multi/test_names.py]: Came from #19029 through #21423

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @h-vetinari

**Pull request merged**